src/Queue/CoverUploadProcessor.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * @file
+ * Upload image service queue processor.
+ */
+
+namespace App\Queue;
+
+use App\Entity\Source;
+use App\Entity\Vendor;
+use App\Event\VendorEvent;
+use App\Utils\Message\CoverUploadProcessMessage;
+use App\Utils\Types\VendorState;
+use Doctrine\ORM\EntityManagerInterface;
+use Enqueue\Client\TopicSubscriberInterface;
+use Interop\Queue\Context;
+use Interop\Queue\Message;
+use Interop\Queue\Processor;
+use Karriere\JsonDecoder\JsonDecoder;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Class CoverUploadProcessor.
+ */
+class CoverUploadProcessor implements Processor, TopicSubscriberInterface
+{
+    private $em;
+    private $dispatcher;
+    private $statsLogger;
+
+    private const VENDOR_ID = 11;
+
+    /**
+     * CoverUploadProcessor constructor.
+     *
+     * @param EntityManagerInterface $entityManager
+     * @param LoggerInterface $statsLogger
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(EntityManagerInterface $entityManager, LoggerInterface $statsLogger, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->em = $entityManager;
+        $this->statsLogger = $statsLogger;
+        $this->dispatcher = $eventDispatcher;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Message $message, Context $session)
+    {
+        $jsonDecoder = new JsonDecoder(true);
+        /** @var CoverUploadProcessMessage $uploadProcessMessage */
+        $uploadProcessMessage = $jsonDecoder->decode($message->getBody(), CoverUploadProcessMessage::class);
+        $identifier = $uploadProcessMessage->getIdentifierType();
+
+        // @TODO: Could this be optimized to only be loaded once!
```

Comment:
Yes in the constructor

Reply:
But will there only be one of these classes or will there be a new processor for each job in the queue? If the latest then it would not be better than this.

Reply:
True. Then just remove the @todo